src/ui/public/doc_table/components/table_row.js

```diff
@@ -92,18 +101,26 @@ module.directive('kbnTableRow', function ($compile) {
           openRowHtml
         ];
 
+        const mapping = indexPattern.fields.byName;
         if (indexPattern.timeFieldName) {
           newHtmls.push(cellTemplate({
             timefield: true,
-            formatted: _displayField(row, indexPattern.timeFieldName)
+            formatted: _displayField(row, indexPattern.timeFieldName),
+            filterable: mapping[indexPattern.timeFieldName].filterable,
+            column: indexPattern.timeFieldName
           }));
         }
 
         $scope.columns.forEach(function (column) {
+          var filterable = mapping[column] ? mapping[column].filterable : false;
+          const flattened = indexPattern.flattenHit(row);
+          filterable = flattened[column] === undefined ? false : filterable;
```

Comment:
Normally I try to avoid ternaries but in this case I think a ternary actually makes the code a little more readable. It also lets us avoid mutating the `filterable` variable:

```javascript
const flattenedRow = indexPattern.flattenHit(row);
const isFilterable = 
  flattenedRow[column] === undefined
  ? false
  : !mapping[column]
  ? false
  : mapping[column].filterable;
```